### ISSUE TITLE
Packaging includes all apps from dist folder

### ISSUE THREAD

**Issue by pascalbayer** [bennymeg/nx-electron#5]
The packaging currently leads to large electron builds (> 1GB with 12 other apps in the nx workspace) since all nx apps from the dist folder are copied to the packaged distributable. 

**Steps to reproduce the behavior:**
1. Setup an nx project with multiple frontend, backend apps and build all applications once
2. Run `nx run app:package --overwrite` or `nx run app:package --overwrite --frontendProject app-ui`

**Expected behavior**
Only the provided `frontendProject` app should be bundled with the packaged distributable. Additionally it would be nice if the packaging would automatically trigger the build of the frontendProject with the correct parameters, e.g. I currently have to run `nx build app-ui --prod --aot --baseHref=./` to make sure that the app can resolve all files correctly.

**Desktop (please complete the following information):**
 - OS: macOS 10.14.6
 - Nx Electron Version: 8.0.0-beta.1

**Comment by bennymeg** [comment]
Thank you for you feedback.

You are right, Only the provided `frontendProject` & `backendProject` apps should be bundled with the packaged distributable. I have not found a consistent way to prune all the other built projects (if exiting). I hope I could resolve it in the next few releases, if you have any ideas on the matter I would love to here them. For now you can ignore them in the `packager.options.json` file. Something like that: `"ignore": ["tools", "libs", "dist/api", "(angular|nx)\\.json"]`.

About the second request, I would appreciate if you could open a separate feature request so we could keep things organized. The sort answer is that since I am supporting various frontend project platforms, I would have to accept only subset of the available parameters (the common ones) which might cause a problem in the future. I completely agree the it is a great feature to have, but I believe further discussion is needed. In the mean time, I think a simple workspace schematic is the best solution.

**Comment by pascalbayer** [comment]
This is only a first draft of how I solved multiple issues in a fork of your repo. I've replaced the contents of your `run` function with the following:

```js
import { build, Configuration, Platform, Arch } from 'electron-builder';

...

async function run(
    options: JsonObject & PackageElectronBuilderOptions,
    context: BuilderContext
): Promise<any> {
    await promisify(writeFile)(
        join(context.workspaceRoot, 'dist', 'apps', options.name, 'index.js'),
        `const Main = require('./${options.name}/main.js');`,
        { encoding: 'utf8' }
    );
    const targets = new Map<Platform, Map<Arch, string[]>>();
    targets.set(
        Platform.MAC.createTarget()
            .entries()
            .next().value[0],
        Platform.MAC.createTarget()
            .entries()
            .next().value[1]
    );
    targets.set(
        Platform.WINDOWS.createTarget()
            .entries()
            .next().value[0],
        Platform.WINDOWS.createTarget()
            .entries()
            .next().value[1]
    );
    await build({
        targets,
        config: {
            win: {
                target: 'portable'
            },
            mac: {
                forceCodeSigning: false,
                target: 'default'
            },
            directories: {
                output: join(context.workspaceRoot, options.out)
            },
            files: [
                '**/package.json',
                {
                    from: `./dist/apps/${options.frontendProject}`,
                    to: options.frontendProject,
                    filter: ['*.*']
                },
                {
                    from: `./dist/apps/${options.name}`,
                    to: options.name,
                    filter: ['main.js']
                },
                {
                    from: `./dist/apps/${options.name}`,
                    to: '',
                    filter: ['index.js']
                }
            ],
            npmRebuild: false,
            asar: true
        }
    });
    return { success: true };
}
```

**It solves the following:**

- Only bundles the electron app and the frontend app provided via `frontendProject`
- Since `electron-packager` does not support to build a single executable as output I switched to `electron-builder` which has support for that option

**Improvements needed:**

- Re-add configurability via external configuration file
- Use platform param to define build targets
- Allow to provide your own `package.json` file to further reduce the size of the generated executables
- Re-add logging and proper return values

I hope this might help you

**Comment by bennymeg** [comment]
First of all, thank you for contributing!

I will start by saying that I have fixed the issue! Only the provided `frontendProject` & `backendProject` apps are bundled with the packaged distributable. the fix will be available on the next release. That's means we are left with the following requests:

1. You would like that there will be an option to package the project into a single executable.
2. Add an option to provide your own `package.json` file to further reduce the size of the generated executables.
3. Automatically trigger the build of the `frontendProject` & `backendProject` before packaging.

**1.** 
I did not knew that electron-builder can also build a single executable, that very interesting.
However, making the change will require a lot of work and I am not sure that's the right way to go.
I have chose to separate the packaging process from the making process (similar to electron-forge) by design. I believe bundling the application into a single package that can then be transformed (using one of these [amazing tools](https://github.com/electron/electron-packager#distributable-creators)) into a distributable package is the right way to go (at least for the time being). 

**2.** 
That could be nice, I will open a new feature request for that, I have some more idea's on that.

**3.** 
I would open a new feature request for that as well, I might create a basic workspace schematic that can be modified by users to their specific project platforms.

What do you think? I that acceptable by you?

**Comment by pascalbayer** [comment]
That sounds great! Thank you for the great work and your efforts in pushing the topic forward.

**Comment by bennymeg** [comment]
I have done some thinking, and researched a little bit more about `electron-builder`.
I wasn't planning to add build in support for making, but now I am thinking to add `electron-builder` as a maker (along side `electron-packager`).

If you wish to participate or have more idea's about the implementation, let me know.

**Comment by pascalbayer** [comment]
Definitely reach out to me if I can support with the implementation.

**Comment by bennymeg** [comment]
@pascalbayer I have implemented a new 'make' schematic (#7).
I would love to here your thoughts about it.

**Comment by pascalbayer** [comment]
@bennymeg I had a look into the new make schematic and upgraded all my apps, looks great and works fine so far. The only thing I wasn't able to figure out yet, was the option to generate a single exe file for the windows platform. It seems like it is always using the 'nsis' target (which builds a single installer/setup exe file) for platform 'win' instead of the 'portable' (standalone exe file with no further dependencies) one. Maybe you can point me in a direction to solve this.
